pkg/package-server/provider/registry.go

```diff
@@ -189,7 +189,18 @@ func (p *RegistryProvider) syncCatalogSource(obj interface{}) (syncError error)
 		Namespace: source.GetNamespace(),
 		Name:      source.GetName(),
 	}
+
 	if sourceMeta := p.sources.GetMeta(key); sourceMeta != nil && sourceMeta.Address == address {
+		logger.Infof("updating PackageManifest based on CatalogSource changes: %v", key)
+		timeout, cancel := context.WithTimeout(context.Background(), cacheTimeout)
+		defer cancel()
+		client, err := p.registryClient(key)
+		if err == nil {
+			syncError = p.refreshCache(timeout, client)
+		} else {
+			syncError = err
+		}
```

Comment:
```suggestion
		if err!= nil {
			syncError = err
			return
		}
	syncError = p.refreshCache(timeout, client)
```

Reply:
@Bowenislandsong you should exit earlier rather than using an `if/else`

Reply:
You know they take exactly the same number of steps! lol

But i see your point of less indent from the if statement.